src/screens/volunteer/VolunteerPersonalScreen.tsx

```diff
@@ -30,6 +32,13 @@ export const VolunteerPersonalScreen = ({ navigation }: StackScreenProps<any>) =
 
   useFocusEffect(refreshPersonal);
 
+  useEffect(() => {
+    if (!isLoading && !isLoggedIn) {
+      navigation.goBack();
+      navigation.navigate(ScreenName.VolunteerLogin);
```

Comment:
@ardasnturk have you tested that on device? how did you feel about the behavior? any other ideas realizing the correct navigation?

Reply:
Yes, of course I tested it on the device. It's like navigating twice and that's not nice. I think I can recommend the [HDVT-89](https://github.com/ikuseiGmbH/smart-village-app-app/pull/676) for that.

Reply:
How is the back navigation handled there in the PR? i can't see it. The problem here is not to navigate when not logged in but that the navigation needs to replace the current screen while navigating to the login screen, so that navigating back will not end up on the personal screen. please think about a few minutes and maybe do some research, if this can be achieved in a better way.

Reply:
I tried `navigation.replace(ScreenName.VolunteerLogin)` and it creates the same effect as `goBack()`. But I still think it's better than back and forth navigation.

Reply:
i think the same and changed it